### PR TITLE
Integrate Send Chain Swap

### LINKED ISSUE #26: Integrate `Send/Receive Chain Swaps`

### PR DESCRIPTION
**Pull request by erdemyerebasmaz**
Closes #26 

UI is subject to change w/ #6 

--- 

Send Chain Swaps feature of Liquid SDK
- https://github.com/breez/breez-sdk-liquid/pull/298

is integrated with this PR.

### CODE REVIEW

File: lib/routes/chainswap/send/fee/fee_option.dart [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import 'package:flutter_breez_liquid/flutter_breez_liquid.dart';
+
+extension PreparePayOnchainResponseAffordable on PreparePayOnchainResponse {
+  bool isAffordable({int? balance, int? walletBalance, required int amountSat}) {
```
Comment (dangeross) [inline lib/routes/chainswap/send/fee/fee_option.dart:4]: Are `walletBalance` and `amountSat` needed?

File: lib/routes/chainswap/send/send_chainswap_form_page.dart [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+    int amount = 0;
+    try {
+      amount = widget.bitcoinCurrency.parse(_amountController.text);
+    } catch (e) {
+      _log.warning("Failed to parse the input amount", e);
+    }
+    return amount;
+  }
+
+  void _prepareReverseSwap() async {
```
Comment (dangeross) [inline lib/routes/chainswap/send/send_chainswap_form_page.dart:84]: Rename to `_prepareSendChainSwap()`?

File: lib/routes/chainswap/send/send_chainswap_page.dart [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    _fetchReverseSwapPairInfo();
+  }
+
+  void didChangeAppLifecycleState(AppLifecycleState state) {
+    if (state == AppLifecycleState.resumed) {
+      _fetchReverseSwapPairInfo();
+    }
+  }
+
+  Future _fetchReverseSwapPairInfo() async {
```
Comment (dangeross) [inline lib/routes/chainswap/send/send_chainswap_page.dart:36]: Rename to `_fetchOnchainLimits()`?

File: lib/routes/chainswap/send/send_chainswap_confirmation_page.dart [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+                  ? texts.reverse_swap_confirmation_error_funds_fee
+                  : texts.reverse_swap_confirmation_error_fetch_fee,
+            );
+          }
+          if (snapshot.connectionState != ConnectionState.done) {
+            return const Center(child: Loader());
+          }
+
+          if (isAffordable) {
+            return FeeChooser(feeOption: snapshot.data!);
```
Comment (dangeross) [inline lib/routes/chainswap/send/send_chainswap_confirmation_page.dart:63]: I think I need to do another pass to see what `FeeChooser` is doing. Is it just fee confirmation?

Comment (erdemyerebasmaz) [reply lib/routes/chainswap/send/send_chainswap_confirmation_page.dart:63]: Yes, it's just fee confirmation.

It was displaying fee options for economy, regular & priority fee rates on c-breez. It displays fee breakdown of `OnchainPaymentLimitsResponse` on l-breez.

It's not really needed, we can use FeeBreakdown directly. 

This whole UI code will change with #6, that's why it's in a poor state.

File: lib/routes/chainswap/send/fee/fee_option.dart (hunk 1 above)
Comment (erdemyerebasmaz) [reply lib/routes/chainswap/send/fee/fee_option.dart]: They are not, removed.

File: lib/routes/chainswap/send/send_chainswap_confirmation_page.dart (hunk 4 above)
Comment (dangeross) [reply lib/routes/chainswap/send/send_chainswap_confirmation_page.dart]: @erdemyerebasmaz perhaps hold off on this. We are going to add the option for the user to select the fee rate or use the boltz default claim estimated fee.

Comment (erdemyerebasmaz) [reply lib/routes/chainswap/send/send_chainswap_confirmation_page.dart]: > @erdemyerebasmaz perhaps hold off on this. We are going to add the option for the user to select the fee rate or use the boltz default claim estimated fee.

@dangeross Holding off these payment UI PR's is a blocker for new UI changes on #6.
Let's add it once it's implemented on top of new UI 👍

**Review by dangeross — APPROVED** [review]
LGTM

**Pull request merged**